pennylane/interfaces/autograd.py

```diff
@@ -66,269 +102,64 @@ def execute(tapes, device, execute_fn, gradient_fn, gradient_kwargs, _n=1, max_d
     parameters = autograd.builtins.tuple(
         [autograd.builtins.list(t.get_parameters()) for t in tapes]
     )
-    return _execute(
-        parameters,
-        tapes=tapes,
-        device=device,
-        execute_fn=execute_fn,
-        gradient_fn=gradient_fn,
-        gradient_kwargs=gradient_kwargs,
-        _n=_n,
-        max_diff=max_diff,
-    )[0]
+    return _execute(parameters, tuple(tapes), execute_fn, jpc)
 
 
 @autograd.extend.primitive
 def _execute(
     parameters,
-    tapes=None,
-    device=None,
-    execute_fn=None,
-    gradient_fn=None,
-    gradient_kwargs=None,
-    _n=1,
-    max_diff=2,
-):  # pylint: disable=dangerous-default-value,unused-argument
+    tapes,
+    execute_fn,
+    jpc,
+):  # pylint: disable=unused-argument
     """Autodifferentiable wrapper around ``Device.batch_execute``.

```

Comment:
I suppose it's fine to leave `batch_execute` for now since it's barely meaningful here at this point

Reply:
updating.